### DISCUSSION TITLE
Sunbird Admin Module - SB Admin - Operations Hub for Sunbird

### DISCUSSION THREAD

**Discussion by baljeetsingh** [Sunbird-Ed/Community#47] [Contribution]
**We have use case where we want to give users a way to / an ability to**

- Create / Manage Organization on Sunbird Instance
- Create / Manage Users
- Create / Manage Framework(s) , Channel
- Create / Manage forms

**Context**

Right now all of above is done through scripts / curl . Thus every support , implementation and installation needs to know all the nitty gritty of system. 

What we aim to do is a way where one can plugin / configure the administrator for sunbird and start doing the support and implementation much faster. 

We feel This would also enable way towards independent / self serve product and create adaption of sunbird would be more easier.

**Access Control - Who Can use These??**

- User who has org admin access to the particular org should be able to manage the respective org
- User who has org admin access to Custodian org should be able to manage all the orgs of the instance 

**Availability - Weather Part of Portal or Separate Module?**

- We are proposing it to be a separate angular app that can be configured / point to respective sunbird instance as add on capability
- e.g. it can be configured as admin.sunbird.org or sunbird.org/admin

**Repo - Where is code ?**

We have started putting down initial pieces of Org Management, User Management , Bulk User Creation and aim to share for larger benefit soon.

*Please do share your question, comments, input and requirements around this.*

CC: @coolbung , @parthlawate

**Comment by pramodkvarma** [comment]
This is great idea to create an admin UX module. But couple of suggestions. Please do not create a single monolithic admin for Sunbird-ED, rather can we think of micro admin modules/pages created for each Sunbird BB separately (Lern, coKreat, RC, Knowlg, Obsrv, etc). These independent admin modules can assume user auth and role management is done outside. So, a simple wrapper admin UX (with login and user role mapping) can be created separately that can simply add menu items for each BB and embed those pages.

This allows independent building block adopters (like RC etc) to also leverage these admin pages by embedding it into their admin console with user authentication and authorisation.

@coolbung @vrayulu @parthlawate

**Comment by sharathkashyap** [comment]
@baljeetsingh There is a need to have micro front end admin module. Which can be plugged and played along with other building blocks including ED to Configure the tenants, frameworks and form configurations. Super Cool idea ! We can help you with semi automated code snippets used by Implementation team today. This can come in handy for your contribution.

There is also a dedicated JIRA identified in terms of creating self serviceable modules. We can also try contributing to the module as and when needed from ED.

cc: @luckynvdu

**Comment by parthlawate** [comment]
A Micro frontend for each BB is a great approach ! It will allow stitching together integrated solutions based on the use case !

**Comment by kameshbhr** [comment]
Currently we have "Manage" page enabled for users with Admin role of a tenant to assign roles to users, access user info etc. This probably is part of Sunbird-ED portal itself. Like @pramodkvarma suggested, it would be better to have each of these as widgets within each BB.
All user management related UI in Lern, Framework management UI in Knowlg etc. Sunbird-ED portal can assemble them in relevant pages.

One thing to think through: 
In each org, there would be two distinct roles for management - An Org Administrator and a System Administrator. Org Admin is typically from program management or administration team who will manage the business processes. An org admin typically will do actions such as creating users, assigning roles, accessing reports etc. Where as System admin will be a technical person performing configurations of the system - such as framework config, form config etc.
At an instance level, we may only have System admin.

**Reply by baljeetsingh** [reply]
@kameshbhr can you help me understand where and how can I see the "Manage" page.

**Reply by parthlawate** [reply]
@kameshbhr can you help with this ?

**Reply by kameshbhr** [reply]
Login to the portal with Admin role and click on the top right profile icon that shows the list of action menus. It would have "Manage" as one of the options.

![image](https://user-images.githubusercontent.com/17846270/192432348-678ef56e-a06e-4516-9c73-a29ceb6ab4f5.png)

**Reply by parthlawate** [reply]
Thanks @kameshbhr .. @baljeetsingh lets try this out

**Reply by baljeetsingh** [reply]
Manage seems allow only to modify the existing user's role

![image](https://user-images.githubusercontent.com/344965/192510238-e1973820-9041-48ac-a466-ef9808cccc71.png)

**Reply by baljeetsingh** [reply]
@parthlawate @kameshbhr

**Reply by kameshbhr** [reply]
@baljeetsingh Yes. It allows modifying user roles and gives user information (of the users who have given consent to share their information). 
There are no user creation flows. Users have to self register and then can be assigned roles.

@luckynvdu @rhwarrier can probably provide more inputs about this in terms of Sunbird BBs

**Reply by luckynvdu** [reply]
@baljeetsingh We may have to start identifying different micro frontend to be created so here are a couple of modules that can be created independently (contributed back to Sunbird ED)
1. Org management
2. Framework management
3. User management
4. Sunbird ED site builder
5. Sunbird ED form management, Etc.
6. Sunbird ED tenant builder

@kameshbhr

**Reply by baljeetsingh** [reply]
Adding them to backlog primarily starting to use for ED and phase 2 could be move other bb

**Comment by baljeetsingh** [comment]
Glad to share the Release 1.0 of this initiative

https://github.com/orgs/Sunbird-Ed/discussions/623 

@coolbung @parthlawate @manojLondhe @vinod303

**Reply by Krishnaj20** [reply]
thanks @baljeetsingh and the entire team for this contribution. This will add lot of value to the adopters. Thank you!
